worker/worker.go

```diff
@@ -1070,6 +1069,28 @@ func (w *worker) objectsHandlerGET(jc jape.Context) {
 	}
 	rw := rangedResponseWriter{rw: jc.ResponseWriter, defaultStatusCode: status}
 
+	// set Last-Modified header
+	jc.ResponseWriter.Header().Set("Last-Modified", time.Now().UTC().Format(http.TimeFormat)) // TODO: update this when object has a ModTime
```

Comment:
I moved this but it might have to move back to the top? I think there was a nextcloud related reason why we have to set `Last-Modified` right? Do we have to set it for all types of responses (empty dir, entries, object)?

Reply:
I think it's the same as the etags. We only need to set it if we return the resource. We also need to verify the timestamp sent by the client and compare it to the object's similar to how etags are handled.